examples/pos/postagging.py

```diff
@@ -0,0 +1,142 @@
+            total += len(y.data)
+        reporter.report({'correct': correct}, self)
+        reporter.report({'total': total}, self)
+
+        return loss
+
+    def argmax(self, xs):
+        hs = [self.feature(x) for x in xs]
+        return self.crf.argmax(hs)
+
+
+def convert(batch, device):
+    sentences = [
+        chainer.dataset.to_device(device, sentence) for sentence, _ in batch]
+    poses = [chainer.dataset.to_device(device, pos) for _, pos in batch]
+    return tuple(sentences + poses)
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='Chainer example: POS-tagging')
+    parser.add_argument('--batchsize', '-b', type=int, default=30,
+                        help='Number of images in each mini batch')
+    parser.add_argument('--epoch', '-e', type=int, default=20,
+                        help='Number of sweeps over the dataset to train')
+    parser.add_argument('--gpu', '-g', type=int, default=-1,
+                        help='GPU ID (negative value indicates CPU)')
+    parser.add_argument('--out', '-o', default='result',
+                        help='Directory to output the result')
+    parser.add_argument('--resume', '-r', default='',
+                        help='Resume the training from snapshot')
+    args = parser.parse_args()
+
+    vocab = collections.defaultdict(lambda: len(vocab))
+    pos_vocab = collections.defaultdict(lambda: len(pos_vocab))
+
+    # Convert word sequences and pos sequences to integer sequences.
+    nltk.download('brown')
+    data = []
+    for sentence in nltk.corpus.brown.tagged_sents():
+        xs = numpy.array([vocab[lex] for lex, _ in sentence], 'i')
+        ys = numpy.array([pos_vocab[pos] for _, pos in sentence], 'i')
+        data.append((xs, ys))
+
+    print('# of sentences: {}'.format(len(data)))
+    print('# of words: {}'.format(len(vocab)))
+    print('# of pos: {}'.format(len(pos_vocab)))
+
+    model = CRF(len(vocab), len(pos_vocab))
+    if args.gpu >= 0:
+        chainer.cuda.get_device(args.gpu).use()
+        model.to_gpu(args.gpu)
+    optimizer = O.Adam()
+    optimizer.setup(model)
+    optimizer.add_hook(chainer.optimizer.WeightDecay(0.0001))
+
+    test_data, train_data = datasets.split_dataset_random(
+        data, len(data) / 10, seed=0)
+
+    train_iter = chainer.iterators.SerialIterator(train_data, args.batchsize)
+    test_iter = chainer.iterators.SerialIterator(test_data, args.batchsize,
+                                                 repeat=False, shuffle=False)
+    updater = training.StandardUpdater(
+        train_iter, optimizer, converter=convert, device=args.gpu)
+    trainer = training.Trainer(updater, (args.epoch, 'epoch'), out=args.out)
+
+    trainer.extend(extensions.Evaluator(
+        test_iter, model, device=args.gpu, converter=convert))
+    trainer.extend(extensions.LogReport())
+
+    trainer.extend(extensions.MicroAverage(
+        'main/correct', 'main/total', 'main/accuracy'))
```

Comment:
It seems that `main/accuracy` is undefined (at least in my environment it does not output `main/accuracy`).